src/main/resources/yamlschema/PrintRequest.yaml

```diff
@@ -0,0 +1,203 @@
+    type: string
+    enum:
+      - cy
+      - en
+    default: en
+    description: The Voter Authority Certificate language
+  certificateFormat:
+    type: string
+    default: standard
+    description: The Voter Authority Certificate format
+    enum:
+      - standard
+      - braille
+      - largeprint
+  deliveryOption:
+    type: string
+    enum:
+      - standard
+    description: Future proof in case of fast track
+  photo:
+    type: string
+    format: binary
+    description: Path to image in zip file
+  deliveryName:
+    type: string
+    maxLength: 255
+    description: Name to be printed with delivery address
+  deliveryStreet:
+    type: string
+    maxLength: 255
+  deliverypProperty:
+    type: string
+    maxLength: 255
+  deliveryLocality:
+    type: string
+    maxLength: 255
+  deliveryTown:
+    type: string
+    maxLength: 255
+  deliveryArea:
+    type: string
+    maxLength: 255
+  deliveryPostcode:
+    type: string
+    maxLength: 10
+  eroNameEn:
+    type: string
+    maxLength: 255
+    description: ERO name
+  eroPhoneNumberEn:
+    type: string
+    maxLength: 20
+    description: ERO contact phone number
+  eroEmailAddressEn:
+    type: string
+    maxLength: 255
+    format: email
+    description: ERO contact email address
+  eroWebsiteEn:
+    type: string
+    maxLength: 1024
+    description: The ERO website url
+  eroDeliveryStreetEn:
+    type: string
+    maxLength: 255
+  eroDeliveryPropertyEn:
+    type: string
+    maxLength: 255
+  eroDeliveryLocalityEn:
+    type: string
+    maxLength: 255
+  eroDeliveryTownEn:
+    type: string
+    maxLength: 255
+  eroDeliveryAreaEn:
+    type: string
+    maxLength: 255
+  eroDeliveryPostcodeEn:
+    type: string
+    maxLength: 10
+  eroNameCy:
+    type: string
+    maxLength: 255
+    description: ERO name
+  eroPhoneNumberCy:
+    type: string
+    maxLength: 20
+    description: ERO contact phone number
+  eroEmailAddressCy:
+    type: string
+    maxLength: 255
+    format: email
+    description: ERO contact email address
+  eroWebsiteCy:
+    type: string
+    maxLength: 1024
+    description: The ERO website url
+  eroDeliveryStreetCy:
+    type: string
+    maxLength: 255
+  eroDeliverypPropertyCy:
+    type: string
+    maxLength: 255
+  eroDeliveryLocalityCy:
+    type: string
+    maxLength: 255
+  eroDeliveryTownCy:
+    type: string
+    maxLength: 255
+  erodDeliveryAreaCy:
+    type: string
+    maxLength: 255
+  eroDeliveryPostcodeCy:
+    type: string
+    maxLength: 10
+required:
+  - requestId
+  - issuingAuthorityEn
+  - issuingAuthorityCy
+  - issueDate
+  - suggestedExpiryDate
+  - requestDateTime
+  - cardFirstname
+  - cardSurname
+  - cardVersion
+  - cardNumber
+  - certificateLanguage
+  - certificateFormat
+  - deliveryOption
+  - photo
+  - deliveryName
+  - deliveryStreet
+  - deliveryPostcode
+  - eroNameEn
+  - eroPhoneNumberEn
+  - eroEmailAddressEn
+  - eroWebsiteEn
+  - eroDeliveryStreetEn
+  - eroDeliveryPostcodeEn
```

Comment:
I assume this is a CSV, so the header will always include all names and this `required` just relates to the data included in a row

Reply:
Yes, absolutely.
It will be a CSV (well, a PSV actually - pipe separated values, but its the same thing! 😁 ). The header row will include all columns (assuming we have a header row - not sure if thats been discussed yet). This `required` attribute is really just there to define which fields are mandatory or not. The codegen will add these fields to the constructor (its a java pojo rather than a kotlin class with non nullable fields). Its about defining the API contract in respect of field cardinality in the CSV